VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-context-menu/judge-context-menu.component.ts

```diff
@@ -0,0 +1,94 @@
+import {
+    ToggleMuteParticipantEvent,
+    ToggleSpotlightParticipantEvent,
+    LowerParticipantHandEvent,
+    CallWitnessIntoHearingEvent,
+    DismissWitnessFromHearingEvent
+} from 'src/app/shared/models/participant-event';
+import { HearingRole } from '../models/hearing-role-model';
+import { CaseTypeGroup } from '../models/case-type-group';
+
+@Component({
+    selector: 'app-judge-context-menu',
+    templateUrl: './judge-context-menu.component.html',
+    styleUrls: ['./judge-context-menu.component.scss']
+})
+export class JudgeContextMenuComponent {
+    private readonly loggerPrefix = '[JudgeContextMenu] -';
+    participant: PanelModel;
+    isDroppedDown = false;
+
+    @Input() set participantInput(participant: PanelModel) {
+        this.participant = participant;
+    }
+
+    @Output() toggleMuteParticipantEvent = new EventEmitter<ToggleMuteParticipantEvent>();
+    @Output() toggleSpotlightParticipantEvent = new EventEmitter<ToggleSpotlightParticipantEvent>();
+    @Output() lowerParticipantHandEvent = new EventEmitter<LowerParticipantHandEvent>();
+    @Output() callWitnessIntoHearingEvent = new EventEmitter<CallWitnessIntoHearingEvent>();
+    @Output() dismissWitnessFromHearingEvent = new EventEmitter<DismissWitnessFromHearingEvent>();
+
+    constructor(private logger: Logger, private elementRef: ElementRef) {}
+
+    @HostListener('document:click', ['$event'])
+    clickout(event) {
+        if (!this.elementRef.nativeElement.contains(event.target)) {
+            this.isDroppedDown = false;
+        }
+    }
+
+    lowerParticipantHand() {
+        this.logger.debug(`${this.loggerPrefix} Attempting to lower hand`, { participant: this.participant.id });
+        this.lowerParticipantHandEvent.emit(new LowerParticipantHandEvent(this.participant));
+    }
+    toggleSpotlightParticipant() {
+        this.logger.debug(`${this.loggerPrefix} Attempting to toggle spotlight`, { participant: this.participant.id });
+        this.toggleSpotlightParticipantEvent.emit(new ToggleSpotlightParticipantEvent(this.participant));
+    }
+
+    toggleMuteParticipant() {
+        this.logger.debug(`${this.loggerPrefix} Attempting to toggle mute`, { participant: this.participant.id });
+        this.toggleMuteParticipantEvent.emit(new ToggleMuteParticipantEvent(this.participant));
+    }
+
+    callWitnessIntoHearing() {
+        this.logger.debug(`${this.loggerPrefix} Attempting to call witness`, { participant: this.participant.id });
+        this.callWitnessIntoHearingEvent.emit(new CallWitnessIntoHearingEvent(this.participant));
+    }
+
+    dismissWitnessFromHearing() {
+        this.logger.debug(`${this.loggerPrefix} Attempting to dismiss witness`, { participant: this.participant.id });
+        this.dismissWitnessFromHearingEvent.emit(new DismissWitnessFromHearingEvent(this.participant));
+    }
+
+    toggleDropdown() {
+        this.isDroppedDown = !this.isDroppedDown;
```

Comment:
Could you log when the context menu is opened/closed please. thanks

Reply:
sure 👍 thanks